milli/src/update/index_documents/transform.rs

```diff
@@ -91,8 +95,12 @@ impl Transform<'_, '_> {
 
         // We extract the primary key from the first document in
         // the batch if it hasn't already been defined in the index
-        let first = documents.peek().and_then(|r| r.as_ref().ok());
-        let alternative_name = first.and_then(|doc| doc.keys().find(|k| k.contains(DEFAULT_PRIMARY_KEY_NAME)).cloned());
+        let first = match documents.peek().map(Result::as_ref).transpose() {
+            Ok(first) => first,
+            Err(_) => return Err(documents.next().unwrap().unwrap_err().into()),
+
```

Comment:
As we know this is an error we can reduce a little bit the `unwrap`s + `return`s boilerplate.

```suggestion
            Err(_) => documents.next().unwrap()?,
```

Reply:
that is not gonna work, the first branch is a ref, and next returns a owned

Reply:
Ho yeah, you are right, my badu!

Reply:
But please remove this newline :)